lib/floe/workflow.rb

```diff
@@ -155,6 +156,11 @@ def end?
     end
 
     def current_state
+      unless context.state.key?("Name")
+        context.state["Name"] = start_at
+        context.state["Input"] = context.execution["Input"].dup
+        context.execution["StartTime"] = Time.now.utc.iso8601
+      end
```

Comment:
I think asking for the current_state potentially setting it is a bit overloaded, what if we had a `start_workflow!` method which set the workflow execution["StartTime"] and set up the initial state?  Then just either handle if `current_state.nil?` or call `start_workflow!` before any checks to `current_state`

Reply:
made these changes.
you can see that `current_state` which is called a lot in specs needed the `start_workflow` - so a bunch of changes

I created a commit before these changes that reworked tests to have less stubbing and possibly call current_state less. This reduced in the test changes.